controller/tests/array_action/xiv/array_mediator_xiv_tests.py

```diff
@@ -111,6 +111,72 @@ def test_delete_volume_succeeds(self):
         self.mediator.client.cmd.vol_delete = Mock()
         self.mediator.delete_volume("vol-wwn")
 
+    def test_get_snapshot_return_correct_value(self):
+        snap_name = "snap"
+        snap_vol_name = "snap_vol"
+        xcli_snap = self._get_single_snapshot_result_mock(snap_name, snap_vol_name)
+        self.mediator.client.cmd.vol_list.return_value = xcli_snap
+        res = self.mediator.get_snapshot(snap_name)
+        self.assertTrue(res.snapshot_name == snap_name)
+        self.assertTrue(res.volume_name == snap_vol_name)
+
+    def test_get_snapshot_same_name_vol_exists_error(self):
+        snap_name = "snap"
+        snap_vol_name = ""
+        xcli_snap = self._get_single_snapshot_result_mock(snap_name, snap_vol_name)
+        self.mediator.client.cmd.vol_list.return_value = xcli_snap
+        with self.assertRaises(array_errors.SnapshotNameBelongsToVolumeError):
+            self.mediator.get_snapshot(snap_name)
+
+    def test_get_snapshot_returns_illegal_object_name(self):
+        snap_name = "snap"
+        self.mediator.client.cmd.vol_list.side_effect = [xcli_errors.IllegalNameForObjectError("", snap_name, "")]
+        with self.assertRaises(array_errors.IllegalObjectName):
+            self.mediator.get_snapshot(snap_name)
+
+    def test_create_snapshot_succeeds(self):
+        snap_name = "snap"
+        snap_vol_name = "snap_vol"
+        size_in_blocks_string = "10"
+        size_in_bytes = int(size_in_blocks_string) * XIVArrayMediator.BLOCK_SIZE_IN_BYTES
+        xcli_snap = self._get_single_snapshot_result_mock(snap_name, snap_vol_name, snap_capacity=size_in_blocks_string)
+        self.mediator.client.cmd.snapshot_create.return_value = xcli_snap
+        res = self.mediator.create_snapshot(snap_name, snap_vol_name)
+        self.assertTrue(res.snapshot_name == snap_name)
+        self.assertTrue(res.volume_name == snap_vol_name)
+        self.assertTrue(res.capacity_bytes == size_in_bytes)
+        self.assertTrue(res.capacity_bytes == size_in_bytes)
+
+    def test_create_snapshot_return_illegal_name_for_object(self):
+        self._test_create_snapshot_error(xcli_errors.IllegalNameForObjectError, array_errors.IllegalObjectName)
+
+    def test_create_snapshot_return_snapshot_exists_error(self):
+        self._test_create_snapshot_error(xcli_errors.VolumeExistsError, array_errors.SnapshotAlreadyExists)
+
+    def test_create_snapshot_return_volume_does_not_exists_error(self):
+        self._test_create_snapshot_error(xcli_errors.VolumeBadNameError, array_errors.VolumeNotFoundError)
+
+    def test_create_snapshot_return_permission_error(self):
+        self._test_create_snapshot_error(xcli_errors.OperationForbiddenForUserCategoryError,
+                                         array_errors.PermissionDeniedError)
+
+    @patch.object(XIVArrayMediator, "_generate_snapshot_response")
+    def test_create_snapshot_generate_snapshot_response_return_exception(self, response):
```

Comment:
raise_exception (not return).
I don't think it's supposed to raise an exception

Reply:
It may if capacity is nut a number. I took existing test for volume